MantidPlot/src/Mantid/MantidDock.cpp

```diff
@@ -404,8 +404,8 @@ void MantidDockWidget::setItemIcon(QTreeWidgetItem *item,
 */
 void MantidDockWidget::updateTree() {
   // do not update until the counter is zero
-  if (m_updateCount.deref())
-    return;
+  // if (m_updateCount.deref())
+  //   return;
```

Comment:
I agree with Raquel here, we need to keep the updateCount check. This is in place to stop updates happening too frequently during heavy processing.

I guess the issue here is the update is called without the ADS having called `MantidDockWidget::incrementUpdateCount()` so the counts en up out of sync.

I wonder if a simpler way to avoid this would to be separate out the update count part from the actual updating. If we had a new `updateTreeOnADSUpdate()` slot that did:
```c++
void MantidDockWidget::updateTreeOnADSUpdate() {
  // do not update until the counter is zero
  if (m_updateCount.deref())
    return;
updateTree();
}
```

we could remove the update count check from `updateTree` and still just call this from the `ConfigDialog`. We would then just need to update the signal/slot connection on line 139 to use the new method and it should avoid the scenario of the counter going below zero?